[EksdeeeX] iP

EksdeeeX opened the pull request:
# Duke Buddy
> "Kiss the tree and come back. Go" -some encik

Duke Buddy helps you to remember and track tasks to complete so as to prevent you from signing extras.

Award-winning capabilities include:
- Tracks ToDos, Events, and Deadlines.
- Easy to learn and use.
- Is as dependable as a well-oiled SAR21.
- ~Helps you to do guard duty~

Interested to try it out? Follow the steps below.
1. Download it [here](https://github.com/EksdeeeX/ip/releases/download/V0.0.1/ip.jar).
2. Start incorporating it into your life and become 100x happier 😀 

Best part is its **FREE!!** 🤭 

Versions:
- [x] Duke Buddy _(Released V0.0.1)_
- [ ] Duke KorKor _(Coming Soon!)_
- [ ] Duke RideOrDie _(Coming Soon!)_
---
Here's a sneak peak of it's ```main``` method:
```
public static void main(String[] args) throws IOException {
        Ui ui = new Ui();
        Parser parser = new Parser();
        ui.initialise();
        parser.initialise();
        .
        .
        .
}
```

Commit:
Add Gradle support

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
no message

EksdeeeX committed:
added date time recognition

EksdeeeX committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

EksdeeeX committed:
no message

EksdeeeX committed:
fix bug

Mehvin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,217 @@
+        StringBuilder desc = new StringBuilder();
+        boolean isDescripton = true;
+        boolean first = false;
+        boolean second = false;
```
Maybe rename the variables, `first` & `second`, to something similar to `isDescripton` to follow the coding standard?
```suggestion
        boolean isDescripton = true;
        boolean isFirst = false;
        boolean isSecond = false;
```

Mehvin commented on src/main/java/Data.java:
```diff
@@ -0,0 +1,108 @@
+
+    public ArrayList<Task> load() throws FileNotFoundException {
+        Scanner sc = new Scanner(this.saved);
+        ArrayList<Task> list = new ArrayList<>();
```
Perhaps a more specific name can be use for the array list (e.g. tasksList)

Mehvin commented on src/main/java/Data.java:
```diff
@@ -0,0 +1,108 @@
+public class Data {
+    private DateTimeFormatter savedFormat = DateTimeFormatter.ofPattern("dd/MMM/yyyy HHmm");
+    private DateTimeFormatter loadFormat = DateTimeFormatter.ofPattern("yyyy/MM/dd HHmm");
+    private boolean exist;
```
Could use a better naming convention for `exist`

Mehvin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,217 @@
+        boolean second = false;
+
+        for(int i = 1; i < task.length; i++) {
+            if(task[i].equalsIgnoreCase("/from")) {
+                first = true;
+                isDescripton = false;
+                continue;
+            }
+            if(task[i].equalsIgnoreCase("/to")) {
+                second = true;
+                first = false;
+                continue;
+            }
+            if(first) {
+                start.append(task[i] + " ");
+            }
+            if(second) {
+                end.append(task[i] + " ");
+            }
+            if(isDescripton) {
```
Spacing can be added to after each `if` to follow the coding standards
```suggestion
            if (task[i].equalsIgnoreCase("/from")) {
                first = true;
                isDescripton = false;
                continue;
            }
            if (task[i].equalsIgnoreCase("/to")) {
                second = true;
                first = false;
                continue;
            }
            if (first) {
                start.append(task[i] + " ");
            }
            if (second) {
                end.append(task[i] + " ");
            }
            if (isDescripton) {
```

Mehvin reviewed (commented):
Overall code looks good 👍, only a few minor mistakes relating to coding standards. 

Additionally, apart from your code, remember that for each git commit to include a commit message subject (a requirement from the [git conventions](https://se-education.org/guides/conventions/git.html)). 

Keep up the good work and all the best for the future weeks!!!

EksdeeeX committed:
added more oop features

EksdeeeX committed:
made into duke package

EksdeeeX committed:
Merge branch 'add-gradle-support'

EksdeeeX committed:
A-JUnit

EksdeeeX committed:
A-Jar

EksdeeeX committed:
Added javadocs to all methods.

EksdeeeX committed:
Corrected code to follow coding standards.

EksdeeeX committed:
a way to find a task by searching for a keyword.

EksdeeeX committed:
Merge tag 'branch-Level-9'

EksdeeeX committed:
no message

EksdeeeX committed:
Implemented GUI

EksdeeeX committed:
Add Assertions

These assumptions have to be met in order for code to function without error.

In some cases, the code might be errneous and these assertions will help identify them.

Assertions act as a checker for correct code.

Assertions are important.

EksdeeeX committed:
Improve code quality.

Code quality could be improved in some areas.

Code was edited in accordance to code quality standards.

Code quality standards should be followed to ensure uniformity.

Good code quality enables better readability

EksdeeeX committed:
Merge pull request #2 from EksdeeeX/branch-A-Assertions

Add Assertions

EksdeeeX committed:
Merge pull request #3 from EksdeeeX/A-CodeQuality

Improve code quality.

EksdeeeX committed:
Add Extension C-tagging and close window upon bye command

EksdeeeX committed:
Update README.md

EksdeeeX committed:
Update README.md

EksdeeeX committed:
Update README.md

EksdeeeX committed:
Update README.md

EksdeeeX committed:
Update README.md

EksdeeeX committed:
Update README.md.
Improve tagging code.
Add Ui.png

EksdeeeX committed:
Merge commit '805b35995c709fa624831e24a76c60245cb0ad4f'

EksdeeeX committed:
improve code quality

EksdeeeX committed:
improve code quality

EksdeeeX committed:
no message

EksdeeeX committed:
Fix bug

EksdeeeX committed:
Fixed bug
